public/app/plugins/datasource/prometheus/querybuilder/components/MetricsLabelsSection.tsx

```diff
@@ -0,0 +1,259 @@
+import React, { useCallback } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+
+import { PrometheusDatasource } from '../../datasource';
+import { getMetadataString } from '../../language_provider';
+import { promQueryModeller } from '../PromQueryModeller';
+import { regexifyLabelValuesQueryString } from '../shared/parsingUtils';
+import { QueryBuilderLabelFilter } from '../shared/types';
+import { PromVisualQuery } from '../types';
+
+import { LabelFilters } from './LabelFilters';
+import { MetricSelect, PROMETHEUS_QUERY_BUILDER_MAX_RESULTS } from './MetricSelect';
+
+export interface MetricsLabelsSectionProps {
+  query: PromVisualQuery;
+  datasource: PrometheusDatasource;
+  onChange: (update: PromVisualQuery) => void;
+  variableEditor?: boolean;
+  onBlur?: () => void;
+}
+
+export function MetricsLabelsSection({
+  datasource,
+  query,
+  onChange,
+  onBlur,
+  variableEditor,
+}: MetricsLabelsSectionProps) {
+  // fixing the use of 'as' from refactoring
+  // @ts-ignore
```

Comment:
~~I think the type is `QueryBuilderLabelFilter[]`. Do we need `@ts-ignore` here?~~

I see that it is about `LabelFilter.tsx` and specifically it is about `onLabelsChange` function. Could you please check if that's possible to have the proper types? Let's avoid `ts-ignore` as much as we can.

Reply:
@ itsmylife This is what we had before

```
onChange={onChangeLabels as (labelFilters: Array<Partial<QueryBuilderLabelFilter>>) => void}
```
https://github.com/grafana/grafana/pull/69884/files#r1229794280

and 
```
  const onChangeLabels = (labels: QueryBuilderLabelFilter[]) => {
    onChange({ ...query, labels });
  };
```
https://github.com/grafana/grafana/pull/69884/files#r1229795429

When I moved this into a new file, using `onChangeLabels as` was not permitted by prettier. I removed the `as` and using the type `QueryBuilderLabelFilter[]` this caused a type error. 
https://github.com/grafana/grafana/pull/69884/files#r1229803709

I am not sure how to fix this as the function definition is the same in the LabelFilters.tsx component but prettier won't allow this.

Reply:
I do not have any suggestions but I checked it briefly and realized that it is not a simple change to make. So we can leave it like this and address it later on.

Reply:
Can we create an issue with the tech-debt label to address this?

Reply:
I will create the issue once this PR is merged to track this.